worker-core/src/main/java/com/hpe/caf/worker/core/GatedHealthProvider.java

```diff
@@ -67,13 +68,20 @@ protected Result check() throws Exception
                 }
 
                 if (unhealthySet.isEmpty()) {
-                    LOG.debug("Reconnecting the incoming queue as all health checks are passing");
-                    workerQueue.reconnectIncoming();
+                    if(!core.isStarted())
+                    {
+                        LOG.debug("Starting WorkerCore...");
+                        core.start();
+                    }
+                    else
+                    {
+                        LOG.debug("Reconnecting the incoming queue as all health checks are passing");
+                        workerQueue.reconnectIncoming();
+                    }
                 } else {
                     LOG.debug("Disconnecting the incoming queue due to the [{}] health check failing", name);
                     workerQueue.disconnectIncoming();
```

Comment:
Presumably it is safe to call this even if it `core.start();` was never called?

Reply:
Seems to be working fine, here is a scenario when worker core was not started:
`[INFO] 18:04:11.294 worker-example> [01:04:11.293Z #2fd.021 INFO  -            -   ] c.h.c.i.w.e.ExampleWorker: Running healthCheck...
[INFO] 18:04:11.294 worker-example> [01:04:11.293Z #2fd.021 DEBUG -            -   ] c.h.c.i.w.e.ExampleWorker: The Example Worker's Health Check is set to Unhealthy
[INFO] 18:04:11.294 worker-example> [01:04:11.293Z #2fd.021 DEBUG -            -   ] c.h.c.w.c.GatedHealthProvider: Disconnecting the incoming queue due to the [worker] health check failing
[INFO] 18:04:11.294 worker-example> [01:04:11.293Z #2fd.021 DEBUG -            -   ] c.h.c.w.q.r.RabbitWorkerQueue: Disconnecting incoming queues
[INFO] 18:04:11.295 worker-example> [01:04:11.295Z #2fd.021 DEBUG -            -   ] c.h.c.w.c.GatedHealthProvider: Disconnecting the incoming queue due to the [queue] health check failing
[INFO] 18:04:11.295 worker-example> [01:04:11.295Z #2fd.021 DEBUG -            -   ] c.h.c.w.q.r.RabbitWorkerQueue: Disconnecting incoming queues
`